buildSrc/src/main/java/org/elasticsearch/gradle/test/RestApiSpecForTestingPlugin.java

```diff
@@ -0,0 +1,141 @@
+                    Dependency coreSpecDep = project.getDependencies()
+                        .project(Map.of("path", ":rest-api-spec", "configuration", "restSpecCore"));
+
+                    if (BuildParams.isInternal()) {
+                        // source the specs from this project - this is the path for Elasticsearch builds
+                        if (project.getPath().startsWith(":x-pack")) {
+                            Configuration xpackSpecConfig = project.getConfigurations().create("copyRestSpecXpack");
+                            Dependency xpackSpecDep = project.getDependencies()
+                                .project(Map.of("path", ":x-pack:plugin", "configuration", "restSpecXpack"));
+                            logger.info("X-pack rest specs for project [{}] will be copied to the test resources.", project.getPath());
+                            createCopyTask(project, xpackSpecConfig, xpackSpecDep, extension.getIncludesXpackSpec(), apiDir, false);
+                        }
+
+                        logger.info("Rest specs for project [{}] will be copied to the test resources.", project.getPath());
+                        createCopyTask(project, coreSpecConfig, coreSpecDep, extension.getIncludesCoreSpec(), apiDir, false);
+
+                    } else {
+                        // source the specs from the published jar - this is the path plugin developers
+                        logger.info(
+                            "Rest specs for project [{}] will be copied to the test resources from the published jar (version: [{}]).",
+                            project.getPath(),
+                            VersionProperties.getElasticsearch()
+                        );
+                        Dependency coreSpecFromJarDep = project.getDependencies()
+                            .create("org.elasticsearch:rest-api-spec:" + VersionProperties.getElasticsearch());
+
+                        createCopyTask(project, coreSpecConfig, coreSpecFromJarDep, extension.getIncludesCoreSpec(), "", true);
+                    }
+                } else {
+                    logger.info("Rest specs will be ignored for project [{}] since there are no REST tests", project.getPath());
+                    return;
+                }
+            } catch (Exception e) {
+                throw new IllegalStateException("Error configuring the rest-api-spec-for-testing plugin. This is likely a bug.", e);
+            }
+        });
+    }
+
+    private void createCopyTask(Project project, Configuration config, Dependency dep, List<String> includes, String into, boolean isJar) {
+        project.getDependencies().add(config.getName(), dep);
+        Copy copyTask = project.getTasks().create(config.getName() + "Task", Copy.class, copy -> {
+            copy.from(isJar ? project.zipTree(config.getSingleFile()) : config.getSingleFile());
```

Comment:
We need to wrap any calls to `Configuration.getSingleFile()` in some kind of deferrable, such as `Callable` otherwise we'll trigger the configuration to be resolved during configuration time, which in the case of an external dependency means downloading the JAR.

Reply:
Calling getSingleFile() is now part of the task execution.